packages/compiler/core/checker.ts

```diff
@@ -3053,7 +3089,7 @@ export function createChecker(program: Program): Checker {
     declaration: Decorator,
     args: DecoratorArgument[],
     decoratorNode: Node
-  ): boolean {
+  ): [boolean, DecoratorArgument[]] {
```

Comment:
@timotheeguerin We have a test failed with below error, could you take a look?

```

TypeError: key.startsWith is not a function
    at isOpenAPIExtensionKey (file:///home/vsts/work/1/s/common/temp/node_modules/.pnpm/@typespec+openapi@0.44.0_5dsz42v7r7ljwzbfu2zuum3liq/node_modules/@typespec/openapi/src/decorators.ts:75:14)
    at $extension (file:///home/vsts/work/1/s/common/temp/node_modules/.pnpm/@typespec+openapi@0.44.0_5dsz42v7r7ljwzbfu2zuum3liq/node_modules/@typespec/openapi/src/decorators.ts:42:8)
    at applyDecoratorToType (file:///home/vsts/work/1/s/common/temp/node_modules/.pnpm/@typespec+compiler@0.45.0-dev.26/node_modules/@typespec/compiler/core/checker.ts:5480:5)
    at finishTypeForProgramAndChecker (file:///home/vsts/work/1/s/common/temp/node_modules/.pnpm/@typespec+compiler@0.45.0-dev.26/node_modules/@typespec/compiler/core/checker.ts:5456:7)
    at finishType (file:///home/vsts/work/1/s/common/temp/node_modules/.pnpm/@typespec+compiler@0.45.0-dev.26/node_modules/@typespec/compiler/core/checker.ts:3705:12)
    at checkModelProperty (file:///home/vsts/work/1/s/common/temp/node_modules/.pnpm/@typespec+compiler@0.45.0-dev.26/node_modules/@typespec/compiler/core/checker.ts:2993:7)
    at checkModelProperties (file:///home/vsts/work/1/s/common/temp/node_modules/.pnpm/@typespec+compiler@0.45.0-dev.26/node_modules/@typespec/compiler/core/checker.ts:2481:25)
    at checkModelStatement (file:///home/vsts/work/1/s/common/temp/node_modules/.pnpm/@typespec+compiler@0.45.0-dev.26/node_modules/@typespec/compiler/core/checker.ts:2393:5)
    at checkModel (file:///home/vsts/work/1/s/common/temp/node_modules/.pnpm/@typespec+compiler@0.45.0-dev.26/node_modules/@typespec/compiler/core/checker.ts:2324:14)
    at getTypeForNode (file:///home/vsts/work/1/s/common/temp/node_modules/.pnpm/@typespec+compiler@0.45.0-dev.26/node_modules/@typespec/compiler/core/checker.ts:615:16)
Error: TypeSpec compiler v0.44.0
````

Reply:
Are you using all the latest dev version or stable version?

Reply:
This happens in our nightly build which always refer the latest dev to check any breaking.

Reply:
Do you have `@typespec/openapi` as an explicit dependency in your package.json? I think it it is not the tool will not bump it and due to how npm resolve things it will not use the dev version.

Reply:
I can't repro with your spec in the [playground](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAdHlwZXNwZWMvcmVzdCI7CtIZdmVyc2lvbmluZ9Ufb3BlbmFwacwcYXp1cmUtdG9vbHMvyCgtxhVjb3Jl3ytjLWF1dG%2FnAI8KdXNpbmcgVHlwZVNwZWMuSHR0cDvQFVJlc3TRFVbpALfIG0HEbC5Db3JlyBJPcGVuQVBJOwoKLy8KLy8gKiogU2VydmljZSBkZXNjcmlwdGlvbiAqKsQgCkBzxhwoewogIHRpdGxlOiAiQ29uZmlkZW50aWFsIExlZGdlcshEIiwKICDnAT46ICIyMDIyLTA1LTEzIiwKfSkKQHVzZUF1dGgoT8QGMsQFPFtNeUZsb3ddPsYfRGVwZW5kZW5jeSjqAL3oANpzLnYxXzBfUHJldmlld18xKeYAoWVyKAogICJ7bOUAjFVyaX3lAIj%2FAKkgIOQA1SAgyTY6IHVybMQYfQopCm5hbWVzcGFjZecBRFNlY3VyaXR5LsxVxlQ7CgpAZG9jKCJEZWZpbmUgdGhlIGF1dGggZmxvdyIpCm1vZGVsIOYA58V0xi7kAfwgb2bNKyAgxBY6IOYBIMQx5AHdLmltcGxpY2l0O8k9xDVvcml6YeQBrlVybNJJ0CI6ICJodHRwczovL2xvZ2luLm1pY3Jvc29mdG9ubGluZS5jb20vY29tbW9uL3YyLjAvb8RCMi%2FISOQCxMh%2BdG9rZdZ2yBrfbtluxUDLanNjb3Blc9Joxhg6IFvJZ2PrAZgt5gHULuUDa8VrLmRlZmF1bHQiXTsKfeoC%2B1BhcmFtZXRlcuwC%2FXPpAv7lAINBIHVuaXF1ZSDGXmZpZXIgZm9y5QHnc3RhdOUBxcQNx3UgSWYgcmV0dXJuZWQgYXMgcGFydMQjYecCg0VudHJ5LCBpdCBpbmRpY2F0ZXPLTGZyb20gd2hpY2jFFWXELSB3YXMgcmVhZC4iKQpzY2FsYXIgVHJhbnNhY%2BQAsElkIGV4dGVuZHMgc3Ry5QP%2B5wC7VGhlIGNvbGxlyChw6wDwZmluacQXxFXmAqNDyyvqARrlArJxdWVyeekBls5YIGnlAJkgzWs%2FOukAjX3rBGxNxWn1AW1yZXNvdXJjZSgiykhzIinnAMtJ7gF9yyP0AMDmALVrZf8AsyAgQHZpc2liaWxpdHkoIuQBY%2FEAye0AyMZMRGV0YWlscyBhYm91dCBh5wH55gGsxFfrAMt06gGn5ADM5gCo6wII6wQQQ29udGVudOUC7eoCT8pW5ACXxiLqAJP5AML%2FAaj0AN%2FrATLbU%2F8DDv8DDv8DDv8DDvMDDiAg6wFi5ADO7QMY5wJ8IFRPRE86IE5vIHdheSB0byBzZXQgIngtbXMtZW51bS7kBewiIHJpZ2h0IG5vd%2BcB5lJlcHJlc%2BUBkmFuIGFzc2lnbmFibGUgcm9sZcR8QGZpeGVkCsRD5wDNVXNlclJvbGXlAdZBZG1pbmlzdHJhdG9y5AZX5AHhcmli5AfqxA9SZWFkZXIs5QJX6gIxdXNlcuoC9vACau0G0ucCIcQy6gME6gCJ9gMExC4gaWQsIGVpdGhlcuQA20FBRCBvYmplY3QgSUQg5ANdZXLkAdrkAY0gZmluZ2VycHJpbnTnAzLlBGzkB44o5gFDc2tpcC11cmwtZW5jb2TkCTQsIHRydWX6A1%2FkAITwAnrtAKEnc%2BcBcGVk6QFuIMkT5AFiOu8BcusDnsRF6AMq5gJqcmVjZWlw5QDA5gElUsYR6QNy5AEqLy8gRWxpZGVk5QLP5AIH%2FwI8%2FAI8U%2BgDCukEHeUFQfYCNlHEG8Yy5ACJTG9h5AFG6AIjeeYCIsVXQegAy%2BcBo3nkCdLkAxnrAwUgYXTFE%2BQKV%2BQDn2TnA%2FfmAPrrAx%2FnAQXrBILkATrJYecBHMQ7IMgWOvABOu8BvPMEAeoA%2BSDGH%2BgBuuoA9s9F7ADMSUT%2FA%2BX%2FA%2BX%2FAantA%2BXxBMLLdfABuesAgeoBuUNvbW1pdHRlZOQBu1BlbuYBxukBvVJlc3BvbnPkAWPnBQ90byBh5gEg6QVDy3TlAI116wbfz3V15gK49gE5xDznAzfFC%2B0BNP8Bdv8BdusBdsY%2FUGFnaW5hdGVk7AazaWXkBaXnAOlpbiBy6AD%2B6gD1xGVAcGFnZWRSZXN1bHTnAOdQxBLqBh3ESeoA6EFycmF55AGhzmvnBJtpdGVtcwogyBboAmvER3lbXesA8ukDoP8CqfkCqVBh5AqO6waeb%2BQA7nJpZXbmDABuZXh0IOQA3MRlcuUA3%2BgApcQbTGluawogxSbECz%2FkA1nmBgJMb2PlC7I86wC7Pu4JRE9wZXLFJOsJPG91dGUoIi9hcHAi7Ay88gyt5QFMI3N1cOQC%2BHPvDo9jYWRs6w62L3VzZS1zdGFuZGFyZC1vyXciICJUaGlzIGnlBxpleGlzdOQEiecOLCwgd2UgaGF2ZSBhIG5vbi3kC5lvcm3EIslDLiLkAQpzdW1tYXJ55AMb5gE7c%2BQFMGlz5Qgh7Qi4cyDnA6fkAkvkAWDzDdvrCKXrCgnEPGFy5gZXLWNyZeUCqMpc5QYi7gJMxUhvcOUAicpJ5QDu6w6SRm91bmTmAQ8uTm9u5QK16QHEaXN0PMo7PuUCOOkEomFkZOcDGiDlAvH%2FAZD%2FAZD%2FAZD%2FAZDuAZDoAkZlbmNsYXZlUXVvdOcBD%2BoBq0dldHMgccUa5QS1YWxsIG5vZOYNpPcBmucC%2BecGw8U95wCsU0dYIMduIG1lYXN1cmVt5AHmdGhhdCBjYW4gYuUBuuUFK3ZlcmlmecVkdmFsaWRpdOUECeQA1WRlIGFuZCBpdHPITsdzZ2V05gHNZ2V0RewA1vsBzukDXzx7fSwgT2voBbj%2FAcf%2FAcf%2FAcf%2FAcf%2FAcf2Acdnb3Zlcm5hbmNl5A8dc3RpdHXEJfQB0eQBVcwk5gF15AHgykXsAcbnCWTKMOQAtCDmDdXmAb1hc3NlxAPlAZlhcHBs5AXIcHJvcG9zYWxz5gU7xEJvcnRpdW0gbWVtYmVy6AUj7AG3Q89n%2FwG2%2FwG2%2FwG2%2FwG2%2FwG2%2FwG2%2FwG28gG25wEV%2FAGx9QE%2F6A5Pzx7lA1FtYW7kBpb%2BA6rwAYDGSE3HR%2F8BhfkBhf8Baf8Baf8BaSB3aXRo5wFlyD3nBkz3BnXlAUjvCLLlApdh7AZ8Y29y5gi75AnO5gi9cmFuZ%2B0DDkHMMWlkIG3kCKPlEN5hbGx55ATE6QuNLiBPbmx5yXDnBs7JHyAob%2BURm%2BQSqynMVndpbGzESegJU8Z%2F5wam7gkt%2BgGf7QagyjV5LCDkAa%2FyCXjmAbfoAprvEMvsAUtXcmnkBVTyEQr%2FASL1ASLGN3Bvc%2BcChOYHtewRNO4A4ugA1kPFKFdpdGjnFeRQcm925A3RTmFtZTzlFevLReQLpOgWAiDqExhzOvYTBuQUkuQWGSAg5wEV5QDXR2XkD5XLVvgDq%2BwBGPINoewLfGlkLiD%2FAS7sAS7kApLoETnFVss67gqfZmV05xFGdmFsdecCwOYD4%2F8BXHJjZeQOif8BQ%2F8BQ%2FMBQ%2BYEZkDmAPnkEB%2FlDlvrAjZSdW7kAlVjdXN05AOXxSggb27%2FAXHqD%2FHsDybnAYLpEBnkBqPlEo1pY3XkFVDvAYTGWMYQU2XkANx0b3IoIi%2FrATboD0D2ATJBxTvyA2byD4H2AQ7mDf3%2FAQ3%2FAQ3%2FAQ3%2FAQ3%2FAQ3uAQ3yDpz%2FARf6ARfGQO8HuFRoZXLkCjVuJ%2BQAq%2BcP4uYBDGl65xlzxT4g5Az96ANjbW%2FkCk7rAV7%2FAUr%2FA8hlIGN1cnLkCqDlA0MgYXZhaWzlFBPnBcjnFPT%2FATDlB1bGSf8DeuQE1ukDS%2BcBO%2FYDhvUDcsYbyyzvBL9EZWxldOQImuoTTO8A%2BcYi5AEu5RQ45QRS%2FQg7b3AgZMU0xE33ANPGVecCBMQo%2BAVT8wCR5wLU6BT06AFP227lBLvWbOYGMnPkFPN1cGTFC9t7QWTkDkvFe84y5xSyZmllbGTnCYPpBrdPclXFKPsAoeYAhsgp7gCrfQ%3D%3D)

Reply:
Thanks for the info! Let me deep dive if any dependency mismatch.